go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/pulumi/pulumi-docker-build
 
-go 1.23.1
+go 1.24.1
```

Comment:
Go directive is upgraded to v1.24 as `github.com/pulumi/pulumi/sdk/go/pulumi-language-go/v3` requires Go v1.24 ([ref](https://github.com/pulumi/pulumi/blob/5c96e4e6aa181bbc8385a12b38653d2dabd6eedf/sdk/go/pulumi-language-go/go.mod#L3)).